### PR TITLE
Ray version update

### PR DESCRIPTION
**Pull request by blublinsky**
## Why are these changes needed?

Update Ray version to the current one

### CODE REVIEW

**Review by revit13 — APPROVED** [review]

File: kfp/kfp_support_lib/shared_workflow_support/Makefile [hunk 1]
```diff
@@ -52,6 +51,8 @@ venv::	pyproject.toml .check-env
 	pip install --upgrade pip; \
 	cd ../../../data-processing-lib/python && make set-versions && cd -; \
 	pip install -e ../../../data-processing-lib/python; \
+	cd ../../../data-processing-lib/ray && make set-versions && cd -; \
+	pip install -e ../../../data-processing-lib/ray; \
```
Comment (roytman) [inline kfp/kfp_support_lib/shared_workflow_support/Makefile:55]: you don't want to set-versions there?

File: kfp/kfp_ray_components/Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM docker.io/rayproject/ray:2.9.3-py310
+FROM docker.io/rayproject/ray:2.24.0-py310
```
Comment (roytman) [inline kfp/kfp_ray_components/Dockerfile:1]: we need to add a make rule to set ray-based image and version

File: kfp/kfp_support_lib/shared_workflow_support/Makefile (hunk 1 above)
Comment (blublinsky) [reply kfp/kfp_support_lib/shared_workflow_support/Makefile:55]: version of what?

File: kfp/kfp_ray_components/Dockerfile (hunk 2 above)
Comment (blublinsky) [reply kfp/kfp_ray_components/Dockerfile:1]: You are welcome to do another PR once this one is merged. We actually discussed this with @revit13 when we were debugging this

**Review by roytman — APPROVED** [review]
LGTM

**Review by daw3rd — COMMENTED** [review]
LGTM

**Pull request merged**